### ISSUE TITLE
Comparison of prediction for multivariate time series between LSTM and VARX

### ISSUE THREAD

**Issue by Jii111** [GDSC-Ewha-4th/Study-paperReview#39]
이황현, 홍난영, 이태욱. (2022). VARX 모형과 LSTM 알고리즘의 다변량 시계열 예측 비교. 한국데이터정보과학회지, 33(1), 35-47.

## **1. 서론**

-인공지능을 활용한 시계열(순서있는) 예측 알고리즘
   >**RNN(순환 신경망)** : 시점이 길어질수록, 앞의 정보가 뒤로 충분히 전달되지 못해 원활한 학습 X
   >**LSTM**: RNN의 단점 극복 위해 고안
-본 논문에서는 외부 요인의 효과를 반영하는 외생변수들을 입력변수로 포함하는 다변량 시계열 자료를 LSTM에 적용
-전통적으로 위와 같은 자료에 대해 VARX(vector autoregressive with exogenous variables) 모형 사용

## **2. VARX 모형**

-다변량 시계열 자료의 분석에서는 각각 일변량 시계열 자료의 자기상관성과 더불어 서로 다른 시계열 자료의 교차상관성을 먼저 파악한 후, 이를 반영하는 다변량 시계열 모형 적합 → 미래 예측
-VAR 모형에서 확장(외생변수 포함하는 모형)

-VAR 모형: AR모형을 다변량으로 확장
![image](https://user-images.githubusercontent.com/90593855/200171038-04dfec5d-1027-434d-885a-168695ea4456.png)
-VARX 모형(본 연구에서는 일반적인 외생변수, 계절형 더미변수, 개입변수 함께 고려)
![image](https://user-images.githubusercontent.com/90593855/200171081-c6a2f2f6-2dba-4778-a07c-ff5a9818f074.png)

#### **3.1 RNN과 LSTM**
   > **RNN**
     - 순서가 있는 데이터를 다루기 위해 개발된 인공신경망
     - 은닉층의 노드에서 활성화함수로 나온 결과값을 출력층 방향으로 보내고, 다시 은닉층 노드의 다음 계산의 입력으로 보내는 재귀적 활동
     - 장기의존성 문제: 경사 소실 문제로 인해 시점이 길어질수록 앞의 정보 충분히 전달X
     
   > **LSTM**
     -  기존 RNN에 C라는 셀 추가
     - 셀 상태가 다음 시점으로 전달되면서 기존의 상태를 보존해, 장기의존성 문제 해결
     - 은닉층의 메모리 셀 + **삭제 게이트, 입력 게이트, 출력 게이트** = 불필요한 기억 삭제 & 기억해야할 것들 정함
     => 셀 상태 보호 & 제어
     - **삭제 게이트**: sigmoid 함수를 이용해 기존의 정보를 얼마만큼 삭제할지 결정
     - **입력 게이트**: sigmoid 함수와 hyperbolic tangent 함수를 이용해 새로운 정보를 얼마만큼 저장할 것인지 결정
     - **출력 게이트**: 다음 시점으로 보낼 은닉 상태 결정
     - 주로 연속적인 데이터 처리 **데이터를 적절하게 잘라서 구축**

   #### **3.2 초매개변수(hyper-parameter) 선택**
   > -LSTM은 사전에 **hyper-parameter**를 설정해야 함
      -**학습률(learning rate), 윈도우 길이(window size), 최적화(optimization) 방법, 히든유닛(hidden unit) 개수, 드롭아웃(dropout) 비율, epoch 등**
     
  > - 본 연구에서는 최적화 방법으로 Adam 사용(현재까지 가장 좋은 성능)
      - 과적합 막기 위해 조기 멈춤 사용
      - 활성화 함수 = sigmoid 함수
      - epoch = 100
      - 손실함수 = MSE(평균제곱오차)
      - 이외의 값들은 **베이지안 최적화 방법**을 통해 손실 함수를 작게 하는 조합으로 선택
      -**큰 k값을 사용해, 노이즈로 구성된 작은 사이즈의 클러스터를 정상모델로부터 제거해 정상영역 효율적 구성**

## **4. 모의실험**
-3가지 VARX 모형에 대한 모의실험 진행
; 주기가 7인 계절더미 변수만 포함/일반적인 외생변수만 포함/계절더미와 일반적인 외생변수를 모두 포함
-모든 VARX 모형의 차수 =2, window size = 42 설정
-RMSE와 MAPE 이용해 VARX와 LSTM 비교
   > VARX 모형의 예측 성능이 약간 우수하게 나타났으나, LSTM 알고리즘의 예측 성능도 비교적 안정적  
    실험 상황을 고려하였을 때, 실제 자료 분석에서 VARX 모형에 비해 LSTM 알고리즘의 성능이 더 좋게 나올 수도 있음

## **5. 실증 분석**
-배달 주문 통화 건수 데이터 이용
-배달 주문 통화 건수를 계절 더비 변수 이외의 외생변수를 반영하지 않고 VARX 모형과 LSTM 알고리즘을 사용하여 예측력 비교

**5.1 계절 더미변수만 입력변수로 사용한 경우**     
   -주기가 7인 계절 더미변수만을 고려한 다변량 시계열 자료
   -**VARX 모형**: AIC 기준으로 차수 p=2 선택
   -**LSTM 알고리즘**: window size=42
   -중국 음식의 RMSE를 제외한 모든 경우에 **LSTM 알고리즘**이 VARX보다 우수

**5.2 계절 더미변수를 포함한 모든 외생변수를 입력변수로 사용한 경우**     
   -날씨 데이터와 계절, 스포츠 경기, 명절, 크리스마스 이브 등의 개입변수를 외생변수로 포함
   -**VARX 모형**: AIC 기준으로 차수 p=2 선택
   -**LSTM 알고리즘**: window size=42
   -모두 예측 성능이 향상되었으며, 족발을 제외한 모든 경우에 **LSTM 알고리즘**이 VARX보다 예측성능 우수

## **6. 결론**
본 연구에서는 **다변량 시계열 자료에 외생변수를 포함하여 VARX 모형과 LSTM 알고리즘의 예측력 비교 분석**
3가지 경우에 대해 모의실험 진행 → LSTM 알고리즘의 예측력 안정적
실증 분석 → 다변량 시계열 자료에서 LSTM 알고리즘의 예측력 우수
적절한 외생변수를 고려하여 다변량 시계열 자료를 LSTM 알고리즘에 적용하면 우수한 예측력을 얻을 수도 있음 확인 (계절 더미변수와 일반 외생변수를 모두 사용한 경우에 예측 성능 향상)

**Comment by minha62** [comment]
전체적으로 VARX 모형보다 LSTM 알고리즘의 성능이 더 우수했군요!
좋은 내용 감사합니다:)
